packages/gatsby-plugin-sharp/src/index.js

```diff
@@ -11,6 +11,19 @@ const imageminWebp = require(`imagemin-webp`)
 const queue = require(`async/queue`)
 const path = require(`path`)
 
+const imageSizeCache = new Map()
```

Comment:
This cut ~700ms off a number of queries on gatsbyjs.org

Reply:
👍 👍